pkgs/hook/tool/generate.dart

```diff
@@ -0,0 +1,1270 @@
+    return packages[1];
+  }
+  if (packageImports[packages[1]]!.contains(packages[0])) {
+    return packages[0];
+  }
+  throw UnimplementedError();
+}
+
+String importer(List<String> packages) {
+  if (packages.length != 2) {
+    throw UnimplementedError();
+  }
+  if (packageImports[packages[0]]!.contains(packages[1])) {
+    return packages[0];
+  }
+  if (packageImports[packages[1]]!.contains(packages[0])) {
+    return packages[1];
+  }
+  throw UnimplementedError();
+}
+
+void main() {
+  if (rootSchemas.length != rootSchemas2.length) {
+    throw StateError(
+      'Some schemas are not unique, try adding a unique title field.',
+    );
+  }
+
+  for (final packageName in generateFor) {
+    const schemaName = 'shared';
+    final schemaUri = packageUri.resolve(
+      '../$packageName/doc/schema/$schemaName/shared_definitions.schema.json',
+    );
+    final outputUri = packageUri.resolve(
+      '../native_assets_cli/lib/src/$packageName/syntax.g.dart',
+    );
+    generate(
+      JsonSchemas(rootSchemas[schemaUri]!),
+      schemaName,
+      packageName,
+      outputUri,
+    );
+  }
+}
+
+const remapPropertyKeys = <String, String>{
+  // TODO: Fix the casing in the schema.
+  // 'assetsForLinking': 'assets_for_linking_old',
+};
+
+String remapPropertyKey(String name) {
+  return remapPropertyKeys[name] ?? name;
+}
+
+void generate(
+  JsonSchemas schemas,
+  String schemaName,
+  String packageName,
+  Uri outputUri,
+) {
+  final classes = <String>[];
+  for (final definitionKey in schemas.definitionKeys) {
+    final definitionSchemas = schemas.getDefinition(definitionKey);
+    if (definitionSchemas.generateOpenEnum) {
+      classes.add(generateEnumClass(definitionSchemas, name: definitionKey));
+    } else if (definitionSchemas.generateExtension(packageName)) {
+      classes.add(generateExtension(definitionSchemas));
+    } else if (definitionSchemas.generateClass) {
+      classes.add(generateClass(definitionSchemas, name: definitionKey));
+    }
+  }
+
+  final imports = [
+    if (packageName != 'hook') "import '../hook/syntax.g.dart';",
+    "import '../utils/json.dart';",
+  ];
+
+  final output2 = '''
+// Copyright (c) 2025, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// This file is generated, do not edit.
+
+${imports.join()}
+
+${classes.join('\n\n')}
+''';
+  final file = File.fromUri(outputUri);
+  file.createSync(recursive: true);
+  file.writeAsStringSync(output2);
+  Process.runSync(Platform.executable, ['format', outputUri.toFilePath()]);
+}
+
+String generateExtension(JsonSchemas schemas) {
+  final typeName = schemas.className!;
+  final extensionSchemas = schemas.extensionSchemas;
+  final baseSchemas = schemas.baseSchemas;
+  final baseName = baseSchemas.className;
+  if (baseName != typeName) {
```

Comment:
In general for all the `throw UnimplementedError()`s, add comments on what these conditions are and why they are not implemented for the untrained eye (like mine!)